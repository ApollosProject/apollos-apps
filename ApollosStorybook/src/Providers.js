import React from 'react';
import ApollosConfig from '@apollosproject/config';
import { Providers } from '@apollosproject/ui-kit';
import { MediaPlayerProvider } from '@apollosproject/ui-media-player';
import { LiveProvider } from '@apollosproject/ui-connected';
import ClientProvider, { client } from './client';
import customTheme, { customIcons } from './theme';

const AppProviders = (props) => (
  <ClientProvider>
    <MediaPlayerProvider>
        <LiveProvider>
          <Providers
            themeInput={customTheme}
            iconInput={customIcons}
            {...props}
          />
        </LiveProvider>
    </MediaPlayerProvider>
  </ClientProvider>
);

export default AppProviders;
