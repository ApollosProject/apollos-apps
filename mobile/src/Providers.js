import { nest } from 'recompose';

import ClientProvider from '/mobile/client';
import { ThemeProvider } from '/mobile/ui/theme';
import { WebBrowserProvider } from '/mobile/ui/WebBrowser';

export default nest(ClientProvider, ThemeProvider, WebBrowserProvider);
