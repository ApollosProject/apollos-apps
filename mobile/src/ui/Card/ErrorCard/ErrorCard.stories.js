import React from 'react';
import { storiesOf } from '@storybook/react-native';

import ErrorCard from '.';

storiesOf('Card/ErrorCard', module).add('Example', () => (
  <ErrorCard message={'Boom!'} error={'What?'} />
));
