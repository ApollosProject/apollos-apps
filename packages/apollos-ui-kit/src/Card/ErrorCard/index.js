import React from 'react';
import PropTypes from 'prop-types';

import styled from '../../styled';
import { H4, H6 } from '../../typography';
import Card, { CardContent } from '..';

const Header = styled({ textAlign: 'center' }, 'Error.Header')(H4);
const ErrorText = styled({ textAlign: 'center' }, 'Error.Text')(H6);

const Err = ({ message = 'Uh oh!', error }) => {
  let errorMessage;
  if (typeof error !== 'string') {
    if (error && error.message) {
      errorMessage = error.message;
    } else if (error && error.error && typeof error.error === 'string') {
      errorMessage = error.error;
    } else {
      errorMessage = 'An unexpected error occured';
    }
  } else {
    errorMessage = error;
  }

  return (
    <Card>
      <CardContent>
        <Header>{message}</Header>
        <ErrorText>{errorMessage}</ErrorText>
      </CardContent>
    </Card>
  );
};

Err.propTypes = {
  message: PropTypes.string,
  error: PropTypes.oneOfType([PropTypes.string, PropTypes.object]),
};

export default Err;
