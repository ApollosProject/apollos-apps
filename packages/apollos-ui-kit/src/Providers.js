import { nest } from 'recompose';

import { ThemeProvider } from './theme';
import { LayoutProvider } from './LayoutContext';

export default nest(ThemeProvider, LayoutProvider);
