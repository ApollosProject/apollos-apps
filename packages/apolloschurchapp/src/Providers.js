import { nest } from 'recompose';

import ClientProvider from 'apolloschurchapp/src/client';
import { ThemeProvider } from 'apolloschurchapp/src/ui/theme';
import { WebBrowserProvider } from 'apolloschurchapp/src/ui/WebBrowser';

export default nest(ClientProvider, ThemeProvider, WebBrowserProvider);
