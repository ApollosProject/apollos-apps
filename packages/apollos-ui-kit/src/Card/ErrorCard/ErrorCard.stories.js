import React from 'react';
import { storiesOf } from '@apollosproject/ui-storybook';

import ErrorCard from '.';

storiesOf('ui-kit/Card/ErrorCard', module).add('Example', () => <ErrorCard />);
