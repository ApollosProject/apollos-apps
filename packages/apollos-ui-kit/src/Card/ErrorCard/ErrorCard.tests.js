import React from 'react';
import renderer from 'react-test-renderer';

import Providers from '../../Providers';

import ErrorCard from '.';

describe('the Card component', () => {
  it('should render', () => {
    const tree = renderer.create(
      <Providers>
        <ErrorCard message={'Boom!'} error={'What?'} />
      </Providers>
    );

    expect(tree).toMatchSnapshot();
  });
  it('should render with an error message inside an object', () => {
    const errorObject = {
      message: 'You have discovered an error message. Do you open it?',
    };
    const tree = renderer.create(
      <Providers>
        <ErrorCard error={errorObject} />
      </Providers>
    );

    expect(tree).toMatchSnapshot();
  });
  it('should render with an error inside an object', () => {
    const errorObject = {
      error: 'Errors, errors, and more errors',
    };
    const tree = renderer.create(
      <Providers>
        <ErrorCard error={errorObject} />
      </Providers>
    );

    expect(tree).toMatchSnapshot();
  });
});
