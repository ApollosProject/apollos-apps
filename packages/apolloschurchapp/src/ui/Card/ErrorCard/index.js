import React from 'react';
import PropTypes from 'prop-types';

import styled from 'apolloschurchapp/src/ui/styled';
import { H4, H6 } from 'apolloschurchapp/src/ui/typography';
import Card, { CardContent } from 'apolloschurchapp/src/ui/Card';

const Header = styled({ textAlign: 'center' }, 'Error.Header')(H4);
const ErrorText = styled({ textAlign: 'center' }, 'Error.Text')(H6);

const Err = ({ message = 'Uh oh!', error }) => {
  let errorMessage;
  if (typeof error !== 'string') {
    if (error && error.message) {
      errorMessage = error.message;
    } else if (error && error.error && typeof error.error === 'string') {
      errorMessage = error.error;
    } else {
      errorMessage = 'An unexpected error occured';
    }
  } else {
    errorMessage = error;
  }

  return (
    <Card>
      <CardContent>
        <Header>{message}</Header>
        <ErrorText>{errorMessage}</ErrorText>
      </CardContent>
    </Card>
  );
};

Err.propTypes = {
  message: PropTypes.string,
  error: PropTypes.oneOfType([PropTypes.string, PropTypes.object]),
};

export default Err;
