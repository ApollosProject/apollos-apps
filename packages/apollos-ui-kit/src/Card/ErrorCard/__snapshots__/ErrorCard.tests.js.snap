// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the Card component should render 1`] = `
<RCTSafeAreaView
  onLayout={[Function]}
  style={
    Object {
      "bottom": 0,
      "left": 0,
      "position": "absolute",
      "right": 0,
      "top": 0,
    }
  }
>
  <View
    onLayout={[Function]}
    style={
      Object {
        "flex": 1,
      }
    }
  />
  <View
    style={
      Object {
        "bottom": 0,
        "left": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
      }
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "#FFFFFF",
          "borderRadius": 16,
          "marginHorizontal": 16,
          "marginVertical": 12,
          "shadowColor": "rgba(165, 165, 165, 0.30000000000000004)",
          "shadowOffset": Object {
            "height": 1,
            "width": 0,
          },
          "shadowOpacity": 1,
          "shadowRadius": 6,
        }
      }
    >
      <View
        style={
          Object {
            "aspectRatio": undefined,
            "borderRadius": 16,
            "overflow": "hidden",
          }
        }
      >
        <View
          style={
            Object {
              "paddingHorizontal": 16,
              "paddingVertical": 16,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#303030",
                "fontFamily": "InterUI-Bold",
                "fontSize": 16,
                "lineHeight": 23.04,
                "textAlign": "center",
              }
            }
          >
            Boom!
          </Text>
          <Text
            style={
              Object {
                "color": "#505050",
                "fontFamily": "InterUI-Bold",
                "fontSize": 12,
                "lineHeight": 17.28,
                "textAlign": "center",
              }
            }
          >
            What?
          </Text>
        </View>
      </View>
    </View>
  </View>
</RCTSafeAreaView>
`;

exports[`the Card component should render with an error inside an object 1`] = `
<RCTSafeAreaView
  onLayout={[Function]}
  style={
    Object {
      "bottom": 0,
      "left": 0,
      "position": "absolute",
      "right": 0,
      "top": 0,
    }
  }
>
  <View
    onLayout={[Function]}
    style={
      Object {
        "flex": 1,
      }
    }
  />
  <View
    style={
      Object {
        "bottom": 0,
        "left": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
      }
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "#FFFFFF",
          "borderRadius": 16,
          "marginHorizontal": 16,
          "marginVertical": 12,
          "shadowColor": "rgba(165, 165, 165, 0.30000000000000004)",
          "shadowOffset": Object {
            "height": 1,
            "width": 0,
          },
          "shadowOpacity": 1,
          "shadowRadius": 6,
        }
      }
    >
      <View
        style={
          Object {
            "aspectRatio": undefined,
            "borderRadius": 16,
            "overflow": "hidden",
          }
        }
      >
        <View
          style={
            Object {
              "paddingHorizontal": 16,
              "paddingVertical": 16,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#303030",
                "fontFamily": "InterUI-Bold",
                "fontSize": 16,
                "lineHeight": 23.04,
                "textAlign": "center",
              }
            }
          >
            Uh oh!
          </Text>
          <Text
            style={
              Object {
                "color": "#505050",
                "fontFamily": "InterUI-Bold",
                "fontSize": 12,
                "lineHeight": 17.28,
                "textAlign": "center",
              }
            }
          >
            Errors, errors, and more errors
          </Text>
        </View>
      </View>
    </View>
  </View>
</RCTSafeAreaView>
`;

exports[`the Card component should render with an error message inside an object 1`] = `
<RCTSafeAreaView
  onLayout={[Function]}
  style={
    Object {
      "bottom": 0,
      "left": 0,
      "position": "absolute",
      "right": 0,
      "top": 0,
    }
  }
>
  <View
    onLayout={[Function]}
    style={
      Object {
        "flex": 1,
      }
    }
  />
  <View
    style={
      Object {
        "bottom": 0,
        "left": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
      }
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "#FFFFFF",
          "borderRadius": 16,
          "marginHorizontal": 16,
          "marginVertical": 12,
          "shadowColor": "rgba(165, 165, 165, 0.30000000000000004)",
          "shadowOffset": Object {
            "height": 1,
            "width": 0,
          },
          "shadowOpacity": 1,
          "shadowRadius": 6,
        }
      }
    >
      <View
        style={
          Object {
            "aspectRatio": undefined,
            "borderRadius": 16,
            "overflow": "hidden",
          }
        }
      >
        <View
          style={
            Object {
              "paddingHorizontal": 16,
              "paddingVertical": 16,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#303030",
                "fontFamily": "InterUI-Bold",
                "fontSize": 16,
                "lineHeight": 23.04,
                "textAlign": "center",
              }
            }
          >
            Uh oh!
          </Text>
          <Text
            style={
              Object {
                "color": "#505050",
                "fontFamily": "InterUI-Bold",
                "fontSize": 12,
                "lineHeight": 17.28,
                "textAlign": "center",
              }
            }
          >
            You have discovered an error message. Do you open it?
          </Text>
        </View>
      </View>
    </View>
  </View>
</RCTSafeAreaView>
`;
